addons/mail/models/mail_canned_response.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+
+
+class MailCannedResponse(models.Model):
+    """
+    Canned Response: content that will automatically replace the shortcut of your choosing. This content can still be adapted before sending your message.
+    """
+
+    _name = "mail.canned_response"
+    _description = "Canned Response"
+    source = fields.Char(
+        "Shortcut",
+        required=True,
+        index="trigram",
+        help="Canned response that will automatically be substituted with longer content in your messages."
+        " Type ':' followed by the name of your shortcut (e.g. :hello) to use in your messages.",
+    )
+    substitution = fields.Text(
+        "Substitution",
+        required=True,
+        help="Content that will automatically replace the shortcut of your choosing. This content can still be adapted before sending your message.",
+    )
+    description = fields.Char("Description")
+    last_used = fields.Datetime("Last Used", help="Last time this canned_response was used")
+    group_ids = fields.Many2many("res.groups", string="Authorized Groups", groups="mail.canned_response_administrator")
```

Comment:
Why putting a groups on the field ? Not sure reading this field should be prevented if you can read the canned response

Reply:
Only canned_response_admin can change the value of this field.